src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20241212105726_EES5627_AddingLabelToRelease.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Migrations.ContentMigrations
+{
+    /// <inheritdoc />
+    public partial class EES5627_AddingLabelToRelease : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropIndex(
+                name: "IX_Releases_PublicationId_Year_TimePeriodCoverage",
+                table: "Releases");
+
+            migrationBuilder.AlterColumn<string>(
+                name: "Slug",
+                table: "Releases",
+                type: "nvarchar(81)",
+                maxLength: 81,
+                nullable: false,
+                oldClrType: typeof(string),
+                oldType: "nvarchar(30)",
+                oldMaxLength: 30);
+
+            migrationBuilder.AddColumn<string>(
+                name: "Label",
+                table: "Releases",
+                type: "nvarchar(50)",
+                maxLength: 50,
+                nullable: true);
+
+            migrationBuilder.CreateIndex(
+                name: "IX_Releases_PublicationId_Year_TimePeriodCoverage_Label",
+                table: "Releases",
+                columns: new[] { "PublicationId", "Year", "TimePeriodCoverage", "Label" },
+                unique: true,
+                filter: "[Label] IS NOT NULL");
```

Comment:
Out of interest, does this unique constraint still take effect for the combination of PublicationId, Year and TimePeriodCoverage when Label IS null? We'd hope it does! 🙂

Reply:
I would hope so! I'll double check

Reply:
I've fixed this with adding `HasFilter(null)` to the configuration for the index

Reply:
I wasn't sure how the null filter applied by the code `HasFilter(null)` is getting related specifically to the `Label` column.

I think the documentation [here](https://learn.microsoft.com/en-us/ef/core/modeling/indexes?tabs=data-annotations#index-filter) covers this. It says 

> When using the SQL Server provider EF adds an 'IS NOT NULL' filter for all nullable columns that are part of a unique index. To override this convention you can supply a null value.

So I think the answer is that it's probably not exclusive to the `Label` column, and would be applicable to other nullable columns if there were any.

Reply:
That's correct yes, that's also what I found :)